Index: pwdutils-3.2.19/lib/read-files.c
===================================================================
--- pwdutils-3.2.19.orig/lib/read-files.c
+++ pwdutils-3.2.19/lib/read-files.c
@@ -30,8 +30,14 @@
 #include <errno.h>
 #include <fcntl.h>
 #include <nss.h>
-#include <bits/libc-lock.h>
+#include <pthread.h>
 #define __libc_lock_t pthread_mutex_t
+#define __libc_lock_define_initialized(CLASS,NAME) \
+  CLASS __libc_lock_t NAME = PTHREAD_MUTEX_INITIALIZER;
+#define __libc_lock_lock(NAME) \
+  pthread_mutex_lock, (&(NAME))
+#define __libc_lock_unlock(NAME) \
+  pthread_mutex_unlock, (&(NAME))
 
 #include "read-files.h"
 
Index: pwdutils-3.2.19/lib/read-files-group.c
===================================================================
--- pwdutils-3.2.19.orig/lib/read-files-group.c
+++ pwdutils-3.2.19/lib/read-files-group.c
@@ -32,8 +32,14 @@
 #include <errno.h>
 #include <fcntl.h>
 #include <nss.h>
-#include <bits/libc-lock.h>
+#include <pthread.h>
 #define __libc_lock_t pthread_mutex_t
+#define __libc_lock_define_initialized(CLASS,NAME) \
+  CLASS __libc_lock_t NAME = PTHREAD_MUTEX_INITIALIZER;
+#define __libc_lock_lock(NAME) \
+  pthread_mutex_lock, (&(NAME))
+#define __libc_lock_unlock(NAME) \
+  pthread_mutex_unlock, (&(NAME))
 
 #include "read-files.h"
 
